hooks/build

```
@@ -0,0 +1,449 @@
+        set +x
+        echo
+    fi
+    IMAGES_TO_PULL="${AIRFLOW_CI_IMAGE}"
+    if [[ "${AIRFLOW_CONTAINER_SKIP_MAIN_IMAGE}" == "true" ]]; then
+        echo "Skip downloading the main Airflow image"
+    else
+        IMAGES_TO_PULL="${IMAGES_TO_PULL} ${AIRFLOW_IMAGE}"
+    fi
+
+    DOCKER_CACHE_DIRECTIVE=""
+    for IMAGE in ${IMAGES_TO_PULL}
+    do
+        echo
+        echo "Checking whether image ${IMAGE} needs to be pulled."
+        echo
+        PULL_IMAGE="false"
+        if [[ "${AIRFLOW_CONTAINER_FORCE_PULL_IMAGES}" == "true" ]]; then
+            echo
+            echo "Pulling images is forced. Pulling ${IMAGE}"
+            echo
+            PULL_IMAGE="true"
+        else
+            IMAGE_HASH=$(docker images -q ${IMAGE} 2> /dev/null)
+            if [[ "${IMAGE_HASH}" == "" ]]; then
+                echo
+                echo "No image ${IMAGE} locally available. Pulling for the first time."
+                echo
+                PULL_IMAGE="true"
+            else
+                echo
+                echo "Image ${IMAGE} is in local registry (${IMAGE_HASH}). Not pulling it!"
+                echo
+                PULL_IMAGE="false"
+            fi
+        fi
+        if [[ "${PULL_IMAGE}" == "true" ]]; then
+            if [[ "${AIRFLOW_CONTAINER_SKIP_PULLING_AIRFLOW_IMAGES}" == "true" ]]; then
+                echo
+                echo "Skipping pulling image ${IMAGE}"
+                echo
+            else
+                echo
+                set -x
+                docker pull ${IMAGE} || true
+                set +x
+                echo
+            fi
+        fi
+        DOCKER_CACHE_DIRECTIVE="${DOCKER_CACHE_DIRECTIVE} --cache-from ${IMAGE}"
+    done
+
+    echo
+    echo "This build uses Docker cache"
+    echo "Cache directive used: ${DOCKER_CACHE_DIRECTIVE}"
+    echo
+elif [[ "${AIRFLOW_CONTAINER_USE_NO_CACHE}" == "true" ]]; then
+    DOCKER_CACHE_DIRECTIVE="--no-cache"
+    echo
+    echo "Skip cache for builds. Everything will be rebuilt from scratch."
+    echo
+    echo "Cache directive used: ${DOCKER_CACHE_DIRECTIVE}"
+    echo
+else
+    DOCKER_CACHE_DIRECTIVE="--cache-from ${AIRFLOW_CI_IMAGE} --cache-from ${AIRFLOW_IMAGE}"
+    echo
+    echo "Use default cache from locally built images."
+    echo
+    echo "Cache directive used: ${DOCKER_CACHE_DIRECTIVE}"
+    echo
+fi
+
+STAT_BIN=stat
+if [[ "${OSTYPE}" == "darwin"* ]]; then
+    STAT_BIN=gstat
+fi
+# Build id identifying the build uniquely
+BUILD_ID=${BUILD_ID:="local"}
+# Branch name for triggered builds
+BRANCH_NAME=${BRANCH_NAME:="master"}
+
+# directory where "deployment" artifacts should be placed
+DEPLOY_DIR=${AIRFLOW_ROOT}/dist/${BRANCH_NAME}/$(date +%Y-%m-%d)/${BUILD_ID}/${PYTHON_VERSION}
+
+mkdir -p "${DEPLOY_DIR}"
+
+FIXED_PERMISSIONS_FILE=${DEPLOY_DIR}/fixed-permissions.txt
+ALL_FILES_FILE=${DEPLOY_DIR}/all-files.txt
+ALL_FILES_IN_CONTEXT_FILE=${DEPLOY_DIR}/all-files-in-context.txt
+
+mkdir -pv ${DEPLOY_DIR}
```

```suggestion
mkdir -pv "${DEPLOY_DIR}"
```
Double quote to prevent globbing and word splitting.

---

Done